xocto/storage/storage.py

```diff
@@ -568,8 +570,12 @@ def get_key(self, key_path: str, version_id: str | None = None) -> S3Object:
     def get_file_type(self, key_path: str) -> str:
         return self._get_boto_object_for_key(key=key_path).content_type
 
-    def fetch_file(self, key_path: str, version_id: str | None = None) -> StreamingBody:
+    def fetch_file(
+        self, key_path: str, version_id: str | None = None, range: str | None = None
+    ) -> StreamingBody:
         boto_object = self._get_boto_object_for_key(key=key_path, version_id=version_id)
+        if range:
+            return boto_object.get(Range=range)["Body"]
         return boto_object.get()["Body"]
```

Comment:
🐼 I'd simplify it to `boto_object.get(Range=range)["Body"]`

Reply:
I'd changed it to satisfy mypy - shoud I ignore instead?

Reply:
I think its fine, better not to execute that code if range is `None`

Reply:
> I'd changed it to satisfy mypy - shoud I ignore instead?
Better to leave it as it is :)